pkg/scheduler/watchers/membercluster/controller_integration_test.go

```diff
@@ -150,6 +155,112 @@ var _ = Describe("scheduler member cluster source controller", Serial, Ordered,
 		})
 	})
 
+	Context("ready cluster has a non-resource property change", func() {
+		BeforeAll(func() {
+			Consistently(noKeyEnqueuedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Workqueue is not empty")
+
+			// Retrieve the cluster.
+			memberCluster := &clusterv1beta1.MemberCluster{}
+			Expect(hubClient.Get(ctx, types.NamespacedName{Name: clusterName1}, memberCluster)).To(Succeed(), "Failed to get member cluster")
+
+			// Update the list of non-resource properties.
+			memberCluster.Status.Properties = map[clusterv1beta1.PropertyName]clusterv1beta1.PropertyValue{
+				dummyNonResourcePropertyName: {
+					Value:           dummyNonResourcePropertyValue,
+					ObservationTime: metav1.NewTime(time.Now()),
+				},
+			}
+			Expect(hubClient.Status().Update(ctx, memberCluster)).Should(Succeed(), "Failed to update member cluster non-resource properties")
+		})
+
+		It("should enqueue CRPs (case 1a)", func() {
+			Eventually(qualifiedKeysEnqueuedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Keys are not enqueued as expected")
+			Consistently(qualifiedKeysEnqueuedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Keys are not enqueued as expected")
+		})
+
+		AfterAll(func() {
+			keyCollector.Reset()
+		})
+	})
+
+	Context("ready cluster has a total capacity change", func() {
+		BeforeAll(func() {
+			Consistently(noKeyEnqueuedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Workqueue is not empty")
+
+			// Retrieve the cluster.
+			memberCluster := &clusterv1beta1.MemberCluster{}
+			Expect(hubClient.Get(ctx, types.NamespacedName{Name: clusterName1}, memberCluster)).To(Succeed(), "Failed to get member cluster")
+
+			// Update the list of non-resource properties.
+			memberCluster.Status.ResourceUsage.Capacity = corev1.ResourceList{
```

Comment:
can we add a test if the capacity value has been changed?

another test to test if the order of resource list matters.

Reply:
For the first one, why not. For the second one, resource list is actually an alias for a map.